service/src/test/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsServiceTest.java

```diff
@@ -0,0 +1,254 @@
+                DocumentSocialWorkOther::getDateTimeUploaded,
+                DocumentSocialWorkOther::getUploadedBy)
+            .containsExactly(
+                "/new_test.doc",
+                "New Additional Doc 1",
+                time.now(),
+                USER);
+    }
+
+    @Test
+    void shouldUpdateOtherSocialWorkDocumentsListWithNewDocument() {
+        List<DocumentSocialWorkOther> updatedDocs = unwrapElements(
+            uploadDocumentsService.setUpdatedByAndDateAndTimeOnDocuments(
+                createCaseDataWithCurrentDocumentSocialWorkOther(),
+                createCaseDataWithOldDocumentSocialWorkOther()));
+
+        //Below tests old updated document title and title for new document
+        assertThat(updatedDocs)
+            .extracting(DocumentSocialWorkOther::getDocumentTitle)
+            .containsExactly("Additional Doc 1 - changed", "Additional Doc 2");
+
+        assertThat(updatedDocs)
+            .extracting(DocumentSocialWorkOther::getTypeOfDocument)
+            .extracting(DocumentReference::getUrl)
+            .containsExactly("/test1 - changed.doc", "/test2.doc");
+    }
+
+    @Test
+    void shouldUpdateDocumentWithUpdatedDetailsAndUser() {
+        Document document =
+            uploadDocumentsService.setUpdatedByAndDateAndTimeOnDocuments(
+                createCaseDataWithUpdatedDocument(),
+                createCaseDataWithOldDocument());
+
+        assertThat(document)
+            .extracting(Document::getTypeOfDocument)
+            .extracting(DocumentReference::getUrl)
+            .isEqualTo("/new_test.doc");
+
+        assertThat(document)
+            .extracting(
+                Document::getDateTimeUploaded,
+                Document::getUploadedBy)
+            .containsExactly(
+                time.now(),
+                USER
+            );
+    }
+
+    @Test
+    void shouldReturnNullWhenCourtBundleDocumentIsNotAttached() {
+        Document document =
+            uploadDocumentsService.setUpdatedByAndDateAndTimeOnDocuments(
+                null,
+                createCaseDataWithOldDocument());
+
+        assertThat(document).isNull();
+    }
+
+    @Test
+    void shouldReturnCurrentDocumentWithUpdatedByAndTimeWhenNoPreviousDocs() {
+        Document document =
+            uploadDocumentsService.setUpdatedByAndDateAndTimeOnDocuments(
+                createCaseDataWithUpdatedDocument(),
+                null);
+
+        assertThat(document)
+            .extracting(Document::getTypeOfDocument)
+            .extracting(DocumentReference::getUrl)
+            .isEqualTo("/new_test.doc");
+
+        assertThat(document)
+            .extracting(
+                Document::getDateTimeUploaded,
+                Document::getUploadedBy)
+            .containsExactly(
+                time.now(),
+                USER
+            );
+    }
+
+    private List<Element<DocumentSocialWorkOther>> createCaseDataWithCurrentDocumentSocialWorkOther() {
```

Comment:
Name could be better, i'd expect method to return CaseData from it's name which it does not do. Same with others below.